src/resolve-config-editorconfig.js

```diff
@@ -0,0 +1,55 @@
+"use strict";
+
+const editorconfig = require("editorconfig");
+const mem = require("mem");
+
+const editorconfigAsyncNoCache = (filePath, config) =>
+  Promise.resolve(
+    filePath &&
+      !config &&
+      editorconfig.parse(filePath).then(editorConfigToPrettier)
+  );
+const editorconfigAsyncWithCache = mem(editorconfigAsyncNoCache);
+
+const editorconfigSyncNoCache = (filePath, config) =>
```

Comment:
Is there any way to follow the same pattern as `getExplorerMemoized()` to prevent repetition?

Reply:
`editorconfig` doesn't allow passing a `sync` parameter like `cosmiconfig` does, so I think we need to have the repetition here, if I'm understanding you correctly.